### PR TITLE
fix missing intrinsics on target type change

### PR DESCRIPTION
**Pull request by JStech**
Fix for a bug I just ran into. If the camera intrinsics were already received, and then you change the target type (ArUco/ChArUco), the new target wouldn't receive the correct intrinsics. The target object is created with default intrinsics (identity camera matrix and all zero distortion), and then the target tab wouldn't send its cached intrinsics to the target object, since they match the incoming CameraInfo message. This PR just sends the intrinsics after the new target object is created.

I also made some more stylistic changes to `cameraInfoCallback`.

### CODE REVIEW

File: moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_target_widget.cpp [hunk 1]
```diff
@@ -434,25 +434,20 @@ void TargetTabWidget::imageCallback(const sensor_msgs::ImageConstPtr& msg)
 
 void TargetTabWidget::cameraInfoCallback(const sensor_msgs::CameraInfoConstPtr& msg)
 {
-  if (target_)
+  if (target_ && msg->height > 0 && msg->width > 0 && !msg->K.empty() && !msg->D.empty() &&
+      (msg->K != camera_info_->K || msg->P != camera_info_->P))
```
Comment (AndyZe) [inline moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_target_widget.cpp:438]: I notice that `msg->R` and `msg->distortion_model` are not checked here. Intentional?

File: moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_target_widget.cpp [hunk 2]
```diff
@@ -434,25 +434,20 @@ void TargetTabWidget::imageCallback(const sensor_msgs::ImageConstPtr& msg)
...
+    camera_info_->header = msg->header;
+    camera_info_->height = msg->height;
+    camera_info_->width = msg->width;
+    camera_info_->distortion_model = msg->distortion_model;
+    camera_info_->D = msg->D;
+    camera_info_->K = msg->K;
+    camera_info_->R = msg->R;
+    camera_info_->P = msg->P;
+    target_->setCameraIntrinsicParams(camera_info_);
+    Q_EMIT cameraInfoChanged(*camera_info_);
```
Comment (AndyZe) [inline moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_target_widget.cpp:450]: Couldn't you set all these `camera_info_` fields at once? Something like,

`camera_info_.reset(msg);`

File: moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_target_widget.cpp (hunk 1 above)
Comment (JStech) [reply moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_target_widget.cpp:438]: `msg->R` is not relevant (it's for stereo cameras). Checking the distortion model (and parameters) would probably be the rigorously correct thing to do, but realistically it's very unlikely that they would change without also changing the camera matrix.

File: moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_target_widget.cpp (hunk 2 above)
Comment (JStech) [reply moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_target_widget.cpp:450]: Yup, good call.

**Review by AndyZe — APPROVED** [review]

**Pull request merged**